src/state.rs

```diff
@@ -247,28 +247,27 @@ impl ProxyState {
         network: Strng,
         source_workload: &Workload,
         addr: SocketAddr,
+        resolution_mode: ServiceResolutionMode,
     ) -> Option<(Arc<Workload>, u16, Option<Arc<Service>>)> {
         if let Some(svc) = self
             .services
             .get_by_vip(&network_addr(network.clone(), addr.ip()))
         {
             // Randomly pick an upstream
             // TODO: do this more efficiently, and not just randomly
```

Comment:
We aren't doing it randomly anymore?

Reply:
It is still random there is just some filtering before we make the random choice